File: python/Queues/Scaler/task_scheduling.py

from collections import deque


def taskScheduling(A, B):
    A = deque(A)
    B = deque(B)

    count = 0
    while A and B:
        if A[0] != B[0]:
            A.append(A.popleft())
        else:
            A.popleft()
            B.popleft()

        count += 1

    return count


print(taskScheduling([2, 3, 1, 5, 4],  [1, 3, 5, 4, 2]))

File: JavaScript/Queue/Scaler/task_scheduling.js
function taskScheduling(A, B) {
  let count = 0;

  while (A.length > 0 && B.length > 0) {
    if (A[0] !== B[0]) {
      A.push(A.shift());
    } else {
      A.shift();
      B.shift();
    }
    count += 1;
  }

  return count;
}

console.log(taskScheduling([2, 3, 1, 5, 4], [1, 3, 5, 4, 2]));
